src/wallet/wallet.h

```diff
@@ -656,6 +656,9 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
     static std::atomic<bool> fFlushScheduled;
     std::atomic<bool> fAbortRescan;
     std::atomic<bool> fScanningWallet;
+    std::mutex mutexScanning;
```

Comment:
Why the mutex when `fScanningWallet` is `std::atomic`? Could use `std::atomic<bool>::compare_exchange_strong`?

Reply:
Yes. Maybe. But lets keep this for now.